src/main/java/com/eternalcode/lobbyheads/head/HeadService.java

```diff
@@ -0,0 +1,139 @@
+        this.hologramPool = new HologramPool(plugin, 50);
+        this.config = config;
+        this.miniMessage = miniMessage;
+        this.server = server;
+    }
+
+    public void createHologram(OfflinePlayer player, HeadInfo headInfo, String headName) {
+        String replace = headName.replace("{PLAYER}", player.getName());
+        String string = PlaceholderAPI.setPlaceholders(player, replace);
+        Component deserialize = this.miniMessage.deserialize(string);
+        String serialize = AdventureLegacy.SECTION_SERIALIZER.serialize(deserialize);
+
+        Hologram hologram = Hologram.builder(this.plugin, PositionAdapter.convert(this.getLocationOffset(headInfo)))
+            .addLine(serialize)
+            .name(this.getHologramName(headInfo))
+            .loadAndBuild(this.hologramPool);
+
+        this.hologramPool.takeCareOf(hologram);
+        this.showHologramToPlayers(hologram);
+
+        this.createHead(headInfo);
+    }
+
+    public void loadHolograms() {
+        String defaultHeadFormat = this.config.head.defaultHeadFormat;
+
+        for (HeadInfo headInfo : this.config.heads) {
+            this.createHologram(this.server.getOfflinePlayer(headInfo.getPlayerUUID()), headInfo, defaultHeadFormat);
+        }
+    }
+
+    public Optional<HeadInfo> find(Position position) {
+        return this.config.heads.stream()
+            .filter(head -> head.getPosition().equals(position))
+            .findAny();
+    }
+
+    public void removeHologram(HeadInfo headInfo) {
+        this.hologramPool.getHolograms().stream()
+            .filter(hologram -> hologram.getName().equals(this.getHologramName(headInfo)))
+            .forEach(this.hologramPool::remove);
+    }
+
+    public void updateHologram(HeadInfo headInfo) {
+        this.removeHologram(headInfo);
+        this.createHologram(this.server.getOfflinePlayer(headInfo.getPlayerUUID()), headInfo, this.config.head.headFormat);
+    }
+
+
+    private void createHead(HeadInfo headInfo) {
+        SkullData skullData = this.skullAPI.awaitSkullData(headInfo.getPlayerName(), 5, TimeUnit.SECONDS);
+        BukkitScheduler scheduler = this.plugin.getServer().getScheduler();
+
+        BlockState blockState = PositionAdapter.convert(headInfo.getPosition()).getBlock().getState();
+        if (blockState instanceof Skull skull) {
+            this.prepareSkullUpdate(scheduler, skullData, skull);
+        }
+    }
+
+    private void prepareSkullUpdate(BukkitScheduler scheduler, SkullData skullData, Skull skull) {
+        scheduler.runTask(this.plugin, () -> this.updateSkull(skullData, skull));
+    }
+
+    private void updateSkull(SkullData skullData, Skull skull) {
+        GameProfile gameProfile = new GameProfile(UUID.randomUUID(), null);
+        gameProfile.getProperties().put(SKULL_TEXTURE_PROPERTY_KEY, new Property(SKULL_TEXTURE_PROPERTY_KEY, skullData.getValue()));
+
+        try {
+            Field profileField = skull.getClass().getDeclaredField("profile");
```

Comment:
Zrób cache dla tego Field'a coś w stylu "singletona"

Reply:
Rozwiń proszę.

Reply:
W/e i tak to nie jest często callowane